tripy/tests/frontend/test_tensor.py

```diff
@@ -204,6 +191,38 @@ def test_multiple_elements_boolean_fails(self):
         with pytest.raises(tp.TripyException):
             bool(tensor)
 
+    @pytest.mark.parametrize(
+        "data",
+        [
+            [[1, 2], [3, 4]],  # from python list
+            np.ones((2, 2), dtype=np.float32),  # from ext tensor
+        ],
+    )
+    def test_explicit_cast(self, data):
+        a = tp.Tensor(data, dtype=tp.float16)
+        assert a.dtype == tp.float16
+
+    @pytest.mark.parametrize(
+        "devices",
+        [
+            ("cpu", "gpu"),
```

Comment:
Yes, that test will better go to `test_copy.py`, because we will be just calling the `copy(self, device)` in Tensor. This is blocked until we fix host allocation.

Reply:
I see what you mean, added tests for `no_explicit_cast` and `no_explicit_copy`, thanks